kloppy/domain/services/event_factory.py

```diff
@@ -66,7 +66,13 @@ def create_event(event_cls: Type[T], **kwargs) -> T:
             f"The following arguments were skipped: {skipped_kwargs}"
         )
 
-    return event_cls(**relevant_kwargs)
+    event = event_cls(**relevant_kwargs)
+
+    statistics = all_kwargs.get("statistics", [])
```

Comment:
I prefer this to be passed via kwargs, a couple of lines above. Like freeze_frame

Reply:
I'm not sure how it would work.

```python
statistics: Optional[List[Statistic]] = field(
        init=False, default_factory=list
    )
```

In my implementation, the statistics cannot be initialized on creation, but only after. Thus, we cannot pass it as kwargs on creation, right?

The reason why I have put `init=False` is because otherwise it complains about `TypeError: non-default argument follows default argument`

Reply:
A problem, however, with my current implementation is that the statistics of an event gets reset to [] when the dataclasses replace() method is being executed on an event as the result of e.g. a transformation. 

Any idea how to properly handle the statistics?

Reply:
@koenvo could you review my latest code? Thanks!